subsys/emds/emds_new.c

```diff
@@ -200,9 +200,18 @@ int emds_store(void)
 
 int emds_clear(void)
 {
+	bool failed = false;
+
 	if (!emds_initialized) {
 		return -ECANCELED;
 	}
 
-	return 0;
+	for (int i = 0; i < PARTITIONS_NUM_MAX; i++) {
+		if (emds_flash_erase_partition(&partition[i])) {
+			LOG_ERR("Failed to erase partition %d:", i);
```

Comment:
Should we add the error code coming from emds_flash_erase_partition. If not there is no use of the different error codes, as the error code is not returned out of emds_clear. I see that there will be 2 prints for the error, one with the error code and then with the partition #. Do we need both error print, or could we merge them?